src/HttpRequestParser.cpp

```diff
@@ -88,23 +101,52 @@ const std::string HttpRequestParser::parseUri(std::string &requestLine)
 {
 	std::string uri;
 	size_t		pos;
+
 	pos = requestLine.find(' ');
 	uri = requestLine.substr(0, pos);
 	requestLine = requestLine.substr(pos, requestLine.length());
+
 	return uri;
 }
 
-const std::string HttpRequestParser::parseHost()
+const std::string HttpRequestParser::getHeaderValue(std::map<std::string, std::string> &headers, std::string toFind)
+{
+	std::map<std::string, std::string>::iterator it;
+	std::string ret = "";
+
+	it = headers.find(toFind);
+	if (it == headers.end())
+		throw BadRequestException("Header not found");
+	else
+		ret = it->second;
+
+	return ret;
+}
+
+void HttpRequestParser::parseHeaders(const std::string &line, std::map<std::string, std::string> &headers)
 {
-	return std::string();
+	size_t pos = line.find(':');
+	if (pos != std::string::npos)
+	{
+		std::string key = line.substr(0, pos);
+		std::string value = line.substr(pos + 1);
+
+		if (headers[key].empty())
+			headers[key] = value;
+	}
+
 }
 
-const std::string HttpRequestParser::parseBody()
+void HttpRequestParser::findBody(std::string newLine, bool &bodyFound)
 {
-	return std::string();
+	if (bodyFound == false && newLine.compare("\r\n"))
```

Comment:
the \r\n comes after each line break so after the request line GET / HTTP/1.1 we get that too! So to me it would be more reliable if we check that when we get \r\n\r\n we know we are at the end of the body:
`POST /cgi-bin/process.cgi HTTP/1.1\r\n
Host: www.tutorialspoint.com\r\n
Content-Type: text/xml; charset=utf-8\r\n
Content-Length: length\r\n
Accept-Language: en-us\r\n
Accept-Encoding: gzip, deflate\r\n
Connection: Keep-Alive**\r\n**
**\r\n**
<?xml version="1.0" encoding="utf-8"?>\r\n
<string xmlns="http://clearforest.com/">string</string>**\r\n**
**\r\n**

Reply:
But this reminds me that I need to trim out the \r from all the header map values, since I have not done that yet.